core/src/stage.cpp

```diff
@@ -634,6 +634,9 @@ void PropagatingEitherWay::computeGeneric(const InterfaceState& start) {
 	else
 		send<dir>(start, InterfaceState(end), std::move(trajectory));
 }
+// Explicitly instantiate templates for both directions
+template void PropagatingEitherWay::computeGeneric<Interface::FORWARD>(const InterfaceState& start);
+template void PropagatingEitherWay::computeGeneric<Interface::BACKWARD>(const InterfaceState& start);
```

Comment:
Why?
The template is private in the class and there should be no need to have a symbol for instantiations outside the object.

Reply:
:thinking: It build without this too, so I removed it for now. @JafarAbdi Do you know if this is necessary?